app/jobs/bsnl_sms_status_job.rb

```diff
@@ -0,0 +1,50 @@
+class BsnlSmsStatusJob
+  include Sidekiq::Worker
+  include Sidekiq::Throttled::Worker
+  BSNL_TIME_ZONE = "Asia/Kolkata"
+
+  sidekiq_options queue: :default
+  sidekiq_options retry: 3
+
+  sidekiq_throttle(
+    threshold: {limit: 250, period: 1.minute}
+  )
+
+  def perform(message_id)
+    response = Messaging::Bsnl::Api.new.get_message_status_report(message_id)
+    handle_api_errors(message_id, response)
+
+    detailable = BsnlDeliveryDetail.find_by!(message_id: message_id)
+    detailable.message_status = response["Message_Status"] if response["Message_Status"]
+    detailable.result = response["Message_Status_Description"] if response["Message_Status_Description"]
+    detailable.delivered_on = parse_timestamp(response["Delivery_Success_Time"]) if response["Delivery_Success_Time"]
+    detailable.save!
+
+    log(response)
+  end
+
+  private
+
+  def log(response)
+    Rails.logger.info(
+      {
+        message_id: response["Message_Id"],
```

Comment:
Do we need a uniform tag in here like `action: bsnl_message_send` or something, so we can isolate these logs easily?

Reply:
That was a terrible name for it, but you get the idea